src/decoding/decoder.rs

```diff
@@ -146,17 +146,20 @@ impl<'ser> Decoder<'ser> {
 
                 let curpos = self.offset;
                 let ival = self.take_int(':')?;
-                let len = usize::from_str_radix(ival, 10).map_err(|_| {
-                    StructureError::SyntaxError(format!("Invalid integer at offset {}", curpos))
-                })?;
+                let len =
+                    usize::from_str_radix(ival, 10).map_err(|_| StructureError::SyntaxError {
```

Comment:
Snafu has a `.context` and `.with_context` method that converts `source` errors from dependencies into the current crate's error type. However, I'm not sure if that should be used here, because `StructureError::SyntaxError` doesn't have a `source` field.

Reply:
Seems like there is no way to use `.context()` or `.with_context()` as long as the target Error doesn't specify a `source`. Without a dedicated `source` the type doesn't implement the required `IntoError` trait.